src/indicators/drought/ecmwf_seasonal.py

```diff
@@ -0,0 +1,301 @@
+            filename += f"_lt{str(leadtime).zfill(2)}d"
+        filename += ".nc"
+        return PROCESSED_DATA_DIR / country_name / ECMWF_SEASONAL_DIR / filename
+
+    def read_processed_dataset(
+        self,
+        country_name: str,
+        country_iso3: str,
+        version: int = DEFAULT_VERSION,
+        leadtime: int = None,
+    ):
+        filepath = self._get_processed_filepath(
+            country_name=country_name,
+            country_iso3=country_iso3,
+            version=version,
+            leadtime=leadtime,
+        )
+        return xr.load_dataset(filepath)
+
+class EcmwfSeasonalForecast(EcmwfSeasonal):
+    def __init__(self):
+        super().__init__(
+            year_min=2000,
+            #TODO: include way to not make it crash with current year (2021)
+            year_max=2020,
+            cds_name="seasonal-monthly-single-levels",
+            dataset="monthly_mean",
+            dataset_variable_name="product_type",
+        )
+
+    def download(
+        self,
+        country_name: str,
+        country_iso3: str,
+        area: Area,
+        # leadtimes: List[int],
+        version: int = DEFAULT_VERSION,
+        split_by_month: bool = True,
+    ):
+        logger.info(
+            f"Downloading ECMWF seasonal forecast v{version} for years {self.year_min} - {self.year_max}"
+        )
+        month_range = range(1, 13) if split_by_month else [None]
+        for year in range(self.year_min, self.year_max + 1):
+            logger.info(f"...{year}")
+
+            for month in month_range:
+                super()._download(
+                    country_name=country_name,
+                    country_iso3=country_iso3,
+                    area=area,
+                    year=year,
+                    month=month,
+                    version=version,
+                )
+
+    def process(
+        self,
+        country_name: str,
+        country_iso3: str,
+        # leadtimes: List[int],
+        version: int = DEFAULT_VERSION,
+    ):
+        logger.info(f"Processing ECMWF Forecast v{version}")
+        # Get list of files to open
+        filepath_list = [
+            self._get_raw_filepath(
+                country_name=country_name,
+                country_iso3=country_iso3,
+                version=version,
+                year=year,
+                month=month,
+            )
+            for year in range(self.year_min, self.year_max + 1)
+            for month in range(1,13)
+        ]
+
+        # Read in all forecasts and combine into one file
+        logger.info(f"Reading in {len(filepath_list)} files")
+        ds = self._read_in_monthly_mean_dataset(filepath_list)
+
+        # Write out the new dataset to a file
+        self._write_to_processed_file(
+            country_name=country_name,
+            country_iso3=country_iso3,
+            ds=ds,
+            version=version,
+        )
+
+#TODO: understand if also the case for ecmwf (copied from glofas)
```

Comment:
You can test this by just calling the xarray expand dims and see if it crashes

Reply:
seems to be working fine for me, so will remove. Didn't dig into this, but since you say it should have been fixed, does it have to do anything with versioning that we should indicate in the requirements?